core/src/main/scala/org/apache/spark/ui/exec/ExecutorThreadDumpPage.scala

```diff
@@ -48,7 +49,13 @@ private[ui] class ExecutorThreadDumpPage(parent: ExecutorsTab) extends WebUIPage
     val time = System.currentTimeMillis()
     val maybeThreadDump = sc.get.getExecutorThreadDump(executorId)
 
-    val content = maybeThreadDump.map { threadDump =>
+    val grepExp = Option(request.getParameter("grepexp")).map(Pattern.compile(_, Pattern.MULTILINE))
+
+    val filteredContent = maybeThreadDump.map { threadDump =>
+      threadDump.filter(thread =>
+        grepExp.map { grep => grep.matcher(thread.stackTrace).find()}.getOrElse(true)
+      )
```

Comment:
this is not a big deal at all, considering its just for a small number of threads, but I realize now that we can pull the Option handling up higher, to completely skip the filter if there is no grep

```scala
grepExp.map { grep =>
  threadDump.filter { thread => grep.matcher(thread.stackTrace).find() }
}.getOrElse(threadDump)
```